configuration/vpn.md

```diff
@@ -56,3 +59,104 @@ Using a ProxyVM to set up a VPN client gives you the ability to:
+
+    In the following, we use the following placeholder:  
+    `$DEV`  For the device created for the connection.  
+    `$PROT` For the protocol used for connection  
+    `$SVR`  For the openvpn server's ip.  
+    `$DNS`  For the dns server's ip.  
+
+
+3.  Setup iptables:  
+    Edit `/rw/config/qubes-firewall-user-script` and add:
+
+    `iptables -P OUTPUT DROP`  
+    This blocks all outgoing traffic, if not specified otherwise.
+    
+    `iptables -I OUTPUT -o $DEV -j ACCEPT`  
+    This allows the local system to connect through the vpn (you dont need this).
+    
+    `iptables -I OUTPUT -o eth0 -d $SVR -p $PROT --dport 1194 -j ACCEPT`  
+    This allows your system to connect to the vpn server with the protocol `$PROT` under the port 1194.
+    
+    `iptables -I OUTPUT -o lo -j ACCEPT`  
+    This allows connections from the system to the system.
+
+    `iptables -I FORWARD -o eth0 -j DROP`  
+    `iptables -I FORWARD -i eth0 -j DROP`  
+    This blocks forwarding of connections through your plain network device (in case the vpn tunnel breaks).
+    
+    `iptables -I FORWARD -o $DEV -j ACCEPT`
+    This allows forwarding of connections through the vpn. (So other AppVMs can use it)
+
+
+    `iptables -t nat -I PR-QBS -p udp --dport 53 -j DNAT --to-destination $DNS`  
+    `iptables -t nat -I PR-QBS -p tcp --dport 53 -j DNAT --to-destination $DNS`  
+    This will rewrite the DNS destination, and the traffic will be routed down the vpn tunnel. (to prevent DNS leaks)
+
+    Now save `/rw/config/qubes-firewall-user-script` and make it executable:  
+    `sudo chmod +x /rw/config/qubes-firewall-user-script`
+    
+4.  Setup the vpn's autostart:  
+    Add `openvpn --config /home/user/vpn.cfg &` to `/rw/config/rc.local` and make it executable  (`sudo chmod +x /rw/config/rc.local`).
+
+    If your ProxyVM takes long to shut down, this may be caused by the vpn connection.  
+    You can fix this by killing the connection on shutdown (issue `killall openvpn` on shutdown).
+    This can be done by adding  
+    
+        echo "[Unit]
+        Description=shutdown
+        Before=shutdown.target reboot.target
+        
+        [Service]
+        RemainAfterExit=yes
+        ExecStart=/bin/true
+        ExecStop=/rw/config/rc.local.shutdown
+        
+        [Install]
+        WantedBy=shutdown.target multi-user.target reboot.target
+        " > /usr/lib/systemd/system/user_shutdown_hook.service;
+
+        systemctl start user_shutdown_hook;
+
+    to `/rw/config/rc.local`.  
+    This creates and starts a service executing the script `/rw/config/rc.local.shutdown` on shutdown.
+    The service will be deleted after each shutdown, since the `root.img` gets reset at each reboot.
+
+    Now create `/rw/config/rc.local.shutdown`, make it executable `chmod +x  /rw/config/rc.local.shutdown` and put the kill command in it:
+
+        #!/bin/bash
+        killall openvpn
```

Comment:
If going to use systemd anyway, better start openvpn itself from systemd. There is even already provided service file (`/usr/lib/systemd/system/openvpn@.service`), which expects config file in `/etc/openvpn/`. So all what is needed should be copying (or better: symlinking) needed files there, then start it: `systemctl --no-block start openvpn@config-name.service`